### PR TITLE
add throughout, latency and serving benchmark for test generation

### PR DESCRIPTION
**Pull request by zhaoyinglia**

### CODE REVIEW

File: benchmarks/test_latency.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+    assert len(model) == 1, "Above condition should have caught this"
+    model = model[0]
+
+    tokenizer = get_tokenizer()
+
+    prompt = ["我" * (args.prompt_len - 1)] * args.micro_batch_size
+
+    print("warming up....")
+    generate_and_post_process(
```
Comment (aoyulong) [inline benchmarks/test_latency.py]: Please add a arg for warmup_iters.

File: benchmarks/test_latency.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    if args.load is not None:
+        _ = load_checkpoint(model, None, None)
+    print("loading model successfully....")
+
+    assert len(model) == 1, "Above condition should have caught this"
+    model = model[0]
+
+    tokenizer = get_tokenizer()
+
+    prompt = ["我" * (args.prompt_len - 1)] * args.micro_batch_size
```
Comment (aoyulong) [inline benchmarks/test_latency.py]: Please add a path to a prompt file, which can include lots of prompts.

File: benchmarks/test_latency.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+    device = next(model.parameters()).device
+    memory_used = torch.cuda.max_memory_allocated(device) / (1024 ** 3)
+    memory_pct = memory_used / (torch.cuda.get_device_properties(device).total_memory / (1024 ** 3)) * 100
+
+    print("="*30)
+    print(f"Batch Size: {args.micro_batch_size}")
+    print(f"Prompt len: {args.prompt_len}")
+    print(f"Generate len: {args.generate_len}")
+    print(f"Avg Latency: {time / args.num_iters:.2f} seconds")
```
Comment (aoyulong) [inline benchmarks/test_latency.py]: Please add more comments about the difference between the latency and throughput calculation.

**Review by aoyulong — COMMENTED** [review]
Please move these benchmarks to a specified folder since they are all related to the inference.

File: benchmarks/test_latency.py (hunk 2 above)
Comment (zhaoyinglia) [reply benchmarks/test_latency.py]: Done.

File: benchmarks/test_latency.py (hunk 1 above)
Comment (zhaoyinglia) [reply benchmarks/test_latency.py]: Warmup num_iters can be 1.

File: benchmarks/test_latency.py (hunk 3 above)
Comment (zhaoyinglia) [reply benchmarks/test_latency.py]: Done.

File: benchmarks/inference/test_latency.py
Comment (aoyulong) [inline benchmarks/inference/test_latency.py:1]: Rename `test` to `benchmark`? These files are not for testing.

**Review by aoyulong — APPROVED** [review]
LGTM

**Pull request merged**